fix clippy error and `cargo check` warnings

hstove opened the pull request:
I recently started seeing Clippy errors due to non-inlined variables in our `build.rs` script. I also fixed the `cargo check` warnings for a few new parts of our test code that are unused.

fdefelici commented:
I was about doing the same, because I noticed this issue is preventing PR to be merged (clippy fails and are removed from the merge-queue.

Anyhow, it seems there are more of them to be addressed now: https://github.com/stacks-network/stacks-core/actions/runs/15928304659/job/44930903003?pr=6231

hstove commented:
It seems that `uninlined-format-args` was added as a default in 1.88, which was released yesterday. I've updated `cargo clippy-stacks` to ignore this, because once I fixed `stacks-common`, there were 300+ for `clarity`, and who knows how many for the other workspaces.
